### PR TITLE
Django 1.8 updates

### PR DESCRIPTION
**Pull request by adamrt**
I noticed this needed to be done in a pull request. 

Please note, while I changed code and updated tests for the 1.8 issue, I don't think I did it in a backwards compatible way. I'm not sure how this is handled in this project that supports multiple Django version.

Any feedback is welcome.

Thank you

### CODE REVIEW

File: vanilla/model_views.py [hunk 1]
```diff
@@ -83,16 +83,11 @@ def get_form_class(self):
         if self.form_class is not None:
             return self.form_class
 
-        if self.model is not None:
-            if self.fields is None:
-                msg = "'Using GenericModelView (base class of '%s) without setting" \
-                    "either 'form_class' or the 'fields' attribute is pending deprecation."
-                warnings.warn(msg % self.__class__.__name__,
-                              PendingDeprecationWarning)
```
Comment (tomchristie) [inline vanilla/model_views.py]: Could you determine which Django version this changes in?
Then we can figure out the right way to document this change and determine if it's okay for us to make the change at this point.

**Comment by adamrt** [comment]
Well there was already a deprecation warning and it was finally removed in 1.8. I assume it was initially deprecated in 1.6 based on Django's normal deprecation policy. I ran the tests and they worked fine on 1.7 and then failed on 1.8 until I made the changes above.

**Comment by tomchristie** [comment]
Okay, think that's good enough for me - let's also have a version bump as part of this pull request.
After that I think we're good to accept this and roll a new release.

**Comment by adamrt** [comment]
Ok, bumped it to 1.0.4!

I tagged it 1.0.4 on this branch as well. 

Thanks

**Comment by jpadilla** [comment]
This looks good to me.

**Comment by tomchristie** [comment]
Thanks! Latest version now on PyPI.

**Pull request merged**